extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/CredentialsProviderConfig.java

```diff
@@ -13,13 +13,51 @@
     /**
      * Database credentials role, as defined by https://www.vaultproject.io/docs/secrets/databases/index.html
      *
-     * One of `database-credentials-role` or `kv-path` needs to be defined. not both.
+     * Only one of `database-credentials-role`, `credentials-role` or `kv-path` can be defined.
      *
+     * @deprecated Use `credentials-role` with `credentials-type` set to `database`
      * @asciidoclet
      */
+    @Deprecated(since = "2.6")
     @ConfigItem
     public Optional<String> databaseCredentialsRole;
 
+    /**
+     * Dynamic credentials' role.
+     *
+     * Roles are defined by the secret engine in use. For example, `database` credentials roles are defined
+     * by the database secrets engine described at https://www.vaultproject.io/docs/secrets/databases/index.html.
+     *
+     * One of `credentials-role` or `kv-path` can to be defined. not both.
+     *
+     * @asciidoclet
+     */
+    @ConfigItem
+    public Optional<String> credentialsRole;
+
+    /**
+     * Type of dynamic credentials, for example `database` or `rabbitmq`.
+     *
+     * Only used when `credentials-role` is defined.
+     *
+     * @asciidoclet
+     */
+    @ConfigItem(defaultValue = "database")
+    public String credentialsType;
```

Comment:
Currently, the way Vault works, you don't need a "type" because all credentials are returned in the exact same format; as it stands I don't see this changing. We could rename this to `credentialsMountPoint` but in Quarkus, and even in Vault vernacular, "type" is used to refer to the plugin via its default mount point, which is the name of the plugin. Additionally, all of the current secrets plugins for credentials support many roles per plugin which negates any real need mount the plugin multiple times in Vault. So `credentialsType` seemed to "fit" to me. In the end, I could really go either way.